persist/fs/options.go

```diff
@@ -112,6 +128,12 @@ func (o *options) Validate() error {
 			"invalid index bloom filter false positive percent, must be >= 0 and <= 1: instead %f",
 			o.indexBloomFilterFalsePositivePercent)
 	}
+	if o.tagEncoderPool == nil {
```

Comment:
Is this guarding against someone doing SetTagEncoderPool(nil)? I don't think we do this check everywhere but seems reasonable